visualize-admin/visualization-tool
# feat: New footnotes / more options / details panel

Closes #1440

This PR introduces quite a few changes related to chart footnotes and metadata (details) panel based on 🎨 [Figma design](https://www.figma.com/design/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?node-id=4851-203884&t=MK2rRDcSPe3RckqJ-4).

### Features
- More options button is now always shown in the edit and layouting steps in the editor and in published mode
- More options button now contains more actions, depending on the chart's state (being edited as sole chart, being edited as one of many charts, being published), including duplicating the chart, copying and editing or sharing a link to the chart
- Metadata panel is now called Details panel
- New design of the chart footnotes
  - Most of the actions were moved to Details panel (more information about the dataset, SPARQL query button, data download)
  - Table view was moved to more options button
  - In case of merged cubes, footnotes per cube will be "smartly" displayed, based on dimensions actually used in the chart, including new legend for combo charts
- Data download menu's behavior is now consistent with other similar menus in the application, needing to be clicked instead of hovered over to open
- All buttons that open Details panel have new styles (blue, without underlines)

### Fixes
- Jumping to edit mode is now possible when editing single URLs layout
- Details panel no longer flies out of the chart container in published mode when being closed
- Details panel can now be used correctly in case of editing multiple charts based on different cubes, instead of always showing the panel for the first chart
- Dimension values in Details panel now load correctly in some rare scenarios for measures in merged cubes mode

### Removals
- Embed styles have been removed, as the new chart footnotes design is already minimalistic 🧹

## How to test
🔎 **To test most of the above features**, go to the following links and take a look at the published charts. You can also copy and edit the chart to holistically test the changes in the editor mode.
- ✅ [PR](https://visualization-tool-git-feat-new-legend-ixt1.vercel.app/en/v/VSIoJa6_L7JZ?dataSource=Prod)
- ❌ [INT](https://int.visualize.admin.ch/en/v/dImaeSXb-lam?dataSource=Prod)

⚠️ **Some of the new features, like new look of the merged cubes footnotes**, can be seen by visiting the links below. You can also copy and edit the charts to e.g. switch to a bar chart and see that the legend won't be displayed twice, as the chart only uses one cube. If you add a segmentation by `Heavy metal`, then a new footnote should appear, as this dimension comes from a different cube.

- ✅ [PR](https://visualization-tool-git-feat-new-legend-ixt1.vercel.app/en/v/tfmjWX5xXRxm?dataSource=Prod)
- ❌ [INT](https://int.visualize.admin.ch/en/v/pjn9XRi0neIt?dataSource=Prod&flag__configurator.add-dataset.new=true&flag__layouter.dashboard.free-canvas=true)
